### DISCUSSION TITLE
Wait node

### DISCUSSION THREAD

**Discussion by swapnil-pando** [imixs/imixs-workflow#787] [General]
Is there a concept of wait node in Imixs workflow. Wait node should wait indefinitely at a point, until some user interaction happens like click of a button. Is there any way to implement this scenario in Imixs?

**Comment by rsoika** [comment]
Hi,
Imixs-Workflow is a event-driven workflow engine. So the behaviour you expect is the default behaviour. A workflow instance is assigned uniquely to a Task which reflects the current state of your process. The workflow engine waits until a new event is triggered by the `processWorkItem` method:

```java
// trigger event with id 10 of the current task
workitem = workflowService.processWorkItem(workitem.event(10));
```
In a web application this is typically the case when the user triggers a workflow action button (generated by the events from the models). To test this behaviour try out the [Imixs-Process-Manager app](https://github.com/imixs/imixs-process-manager). 

The only exceptions are:

 - [timer events](https://www.imixs.org/doc/engine/workflowscheduler.html) - the workflow engine monitors a process instance until a specific due date is reached (this is used e.g. for escalation)
 - [async events](https://www.imixs.org/doc/engine/asyncevents.html) - the workflow engine will trigger a event asynchronous after a defined time delay
